Bacteriophage and their hosts are locked in an age-old arms race. Successful bacteria are subject to predation, forcing the population to diversify, and phage are also quick to adapt tactics for infecting these potential hosts. Sampling of closely related bacterial strains that differ in phage infection profiles can further elucidate the mechanisms of infection. The Polz Lab maintains the Nahant Collection - 243 Vibrio strains challenged by 241 unique phage, all with sequenced genomes. This is the largest phylogenetically resolved host-range cross test available to date. Genetically mapping out the depths of this dataset requires carefully designed analysis techniques as well as further experimental exploration.

First, we narrow in on a specific phage in the Nahant Collection, 2.275.O, to *characterize the pressures that may select for phage that shuttle their own translational machinery*. While translation is generally considered a hallmark of cellular life, some phage carry abundant tRNA. 2.275.O carries 18 tRNA spanning 13 amino acids. We find that while encoding translation-related components requires shuttling a larger phage genome, it also reduces dependence on host translational machinery, allowing the phage to be more aggressive in degrading and recycling the host genome and other resources required for replication.

Next we *develop a systematic approach for uncovering genomic features that underlie phage-host interactions*. We find that correcting for phylogenetic relationships allows us to pick out relevant signals that would otherwise be drowned out by spurious correlations resulting from statistically oversampled blooms of microbes. Using these results, we wrote an interative javascript visualization to facilitate the process of developing testable hypotheses concerning the mechanisms of phage infection and host response. From the visualization, we are able to identify, in the hosts, mobile genetic elements containing restriction modification systems that may defend against infection, as well as membrane protein modifications that may serve as phage attachment sites.
