# Conclusion

Because phage-host studies are generally aimed at identifying many phage that infect a clonal group of hosts, or many hosts that are infected by a small set of phage; up until now, there is perhaps only one dataset [@Moebus1981] that encompasses both polyclonal phage and host populations. The data is not accompanied by any form of molecular information, and so it is limited in its ability to provide insight into the precise mechanisms from which the infection structure arose. The Nahant Collection [@Kauffman2014;@Kauffman2018;@Kauffman2018a] is very unique in being a complete cross test accompanied with sequenced genomes, from which we can derive mechanistic hypotheses; as well as archived samples, with which we can test these hypotheses.

In chapter 2 of this thesis, we focused in on a specific phage in the Nahant Collection, 2.275.O, to characterize the pressures that may select for phage that shuttle their own translational machinery. We were able to conclude that while encoding translation-related components requires shuttling a larger phage genome, it also reduces dependence on host translational machinery, allowing the phage to be more aggressive in degrading and recycling the host genome and other resources required for replication.

This investigation illuminated many other paths that we believe could be fascinating for future exploration. For example, from TGIRT substitutions, we called base modifications, assuming a multinomial distribution of base counts. However, there is a layer of complexity that we were not able to unroll - the distribution could have been a mixture of nascent tRNA and modified species. And in fact, in a tangential exploration examining the unique tRNA reads, we observed a succession pattern in the tRNA species during the course of infection. One question arising from this observation is a methodological one: is it possible to deconvolute the mixture of base substitutions from the mixture of tRNA species by incorporating data from, for example, mass spectroscopy (a gold standard for identifying chemical compounds) or nanopore direct RNA sequencing (this technique has recently been demonstrated [@Garalde2018] and there is some excitement about the potential for direct detection of nucleotide modifications)? If so, we would be able to paint a clearer picture of exactly which tRNA species are present during various stages of the infection. This has applications beyond bacteriophage, as there is currently an ongoing line of research into the role of tRNA through "adaptive translation" during various forms of stress response. Another question that arises is along the same train of thought as "adaptive translation." Do the different species of the same tRNA have different roles? Are some species activated or inactivated by the modifications that are present? Exciting new technologies for probing translation now exist, such as ribosomal footprint profiling [@Ingolia2009], tRNA-ribo-seq [@Chen2018], charge-seq [@Evans2017], etc. Work in combining these techniques with phage growth experiments may be a promising future direction for uncovering further insights into translation.

Another tangential analysis we conducted was to examine the host expression profile throughout infection. While essentially the entire host transcriptome is degraded, we do observe three main outlier groups. It would be interesting to conduct additional trials in order to determine variation around these expression characteristics. And next, it may be of interest to do a closer exploration of the function and structure of these transcripts. For example, one group of outliers consists of transcripts that appear to be degraded more quickly than the average transcript. What makes them susceptible? Are they specifically targetted by the phage for degradation? Another group of outliers consists of transcripts that initially increase in abundance, and then are rapidly degraded. These tend to be stress response genes. Do these genes participate in tuning down phage infection? If they are knocked out, does this result in a more successful infection phenotype (perhaps an increase in burst size or decrease in latent period)? If this experiment is repeated in the same host with different phage, could different stress response pathways be triggered?

In chapter 3, we developed a systematic approach for uncovering genomic features that underlie phage-host interactions. We found that correcting for phylogenetic relationships allows us to pick out relevant signals that would otherwise be drowned out by spurious correlations resulting from statistically oversampled blooms of microbes. Using these results, we wrote an interative javascript visualization to facilitate the process of developing testable hypotheses concerning the mechanisms of phage infection and host response. From the visualization, we are able to identify, in the hosts, mobile genetic elements containing restriction modification systems that may defend against infection, as well as membrane protein modifications that may serve as phage attachment sites.

This analysis was not, by any means, the most obvious path. There were many somewhat promising routes that we wandered down but could find no obvious means for inferring biological insight through. For example, initially we sought to predict the matrix of infections by using the genetic data. One problem that we encountered was that a reasonable model should capture the generally lock-and-key nature of infection specificities due to protein interactions. For example, a specific methylase may evade a specific restriction modification system. However, with 1000 phage protein clusters and 10,000 host protein clusters, this meant we had in total, 10 million possible interaction terms. So there was also a conundrum about whether it was necessary to encode a 60,000 x 10,000,000 matrix of predictors (60,000 being approximately 241 x 243, the dimensions of the infection matrix). Ultimately we were able to utilize a trick called alternating minimization, which is elaborated further in the appendix. Another technique we employed was to conduct an initial step of screening, then using the most promising protein profiles for the final stage of prediction. However, both methods suffered from the same drawback in that the results lacked interpretability. We were able to see some promising results through examining the outer products of the decomposed matrix of interaction coefficients, and this may be a promising direction for future exploration.

Then, of course, because the goal of the visualization we presented in chapter 3 was for the purpose of generating testable hypotheses, there are many hypotheses suggested by this visualization that we hope may eventually be tested. For example infection by phage 1.293.O is very positively associated with the presence of a pyruvyl polysaccharide transferase protein. Could this be a membrane protein modification that serves as an attachment site for the virus? If so, deleting the gene from a host or inserting the gene into a non-host should be able to change the microbes' infection susceptibility. Or, a couple of proteins strongly negatively associated with infection by 1.159.O, 1.028.O, and 1.219.O are annotated as participating in programmed cell death. Could this be a "protect your neighbors" host defense? If so, these genes should be seen to be expressed early in the infection. And finally, there are numerous prophage-like islands and conjugative transfer elements, often associated with various forms of restriction modification systems that are sometimes positively and sometimes negatively associated with infection. This leads us to believe that there may be a network of competitive exclusion (and potentially, cooperation) among these elements. This could be a particularly tricky hypothesis to test, as it may involve making a carefully crafted combinatorial library of various mobile elements.
 
Work on phage-host interactions from the last century has lead to the discovery of important insights into the central dogma as well as the development of impactful technologies such as restriction enzymes, Hfr conjugation, transposon mutagenesis, and CRISPR-Cas [@Salmond2015]. Still, on average approximately 70% of open reading frames in a phage are unannotated hypothetical proteins [@Gonzalez2015]. We are excited to see what future research may unfold.


